Xamarin.Forms.Platform.Android/Renderers/BottomNavigationViewUtils.cs

```diff
@@ -20,6 +30,162 @@ namespace Xamarin.Forms.Platform.Android
 {
 	public static class BottomNavigationViewUtils
 	{
+
+		internal const int MoreTabId = 99;
+
+		public static Drawable CreateItemBackgroundDrawable()
+		{
+			var stateList = ColorStateList.ValueOf(Color.Black.MultiplyAlpha(0.2).ToAndroid());
+			return new RippleDrawable(stateList, new ColorDrawable(AColor.White), null);
+		}
+
+		internal static void UpdateEnabled(bool tabEnabled, IMenuItem menuItem)
+		{
+			if (menuItem.IsEnabled != tabEnabled)
+				menuItem.SetEnabled(tabEnabled);
+		}
+
+		internal static async void SetupMenu(
+			IMenu menu, 
+			int maxBottomItems, 
+			List<(string title, ImageSource icon, bool tabEnabled)> items, 
+			int currentIndex, 
+			BottomNavigationView bottomView, 
+			Context context)
+		{
+			menu.Clear();
+			int numberOfMenuItems = items.Count;
+			bool showMore = numberOfMenuItems > maxBottomItems;
+			int end = showMore ? maxBottomItems - 1 : numberOfMenuItems;
+
+
+			List<IMenuItem> menuItems = new List<IMenuItem>();
+			List<Task> loadTasks = new List<Task>();
+			for (int i = 0; i < end; i++)
+			{
+				var item = items[i];
+				using (var title = new Java.Lang.String(item.title))
+				{
+					var menuItem = menu.Add(0, i, 0, title);
+					menuItems.Add(menuItem);
+					loadTasks.Add(SetMenuItemIcon(menuItem, item.icon, context));
+					UpdateEnabled(item.tabEnabled, menuItem);
+					if (i == currentIndex)
+					{
+						menuItem.SetChecked(true);
+						bottomView.SelectedItemId = i;
+					}
+				}
+			}
+
+			if (showMore)
+			{
+				var moreString = new Java.Lang.String("More");
+				var menuItem = menu.Add(0, MoreTabId, 0, moreString);
+				moreString.Dispose();
+
+				menuItem.SetIcon(Resource.Drawable.abc_ic_menu_overflow_material);
+				if (currentIndex >= maxBottomItems - 1)
+					menuItem.SetChecked(true);
+			}
+
+			bottomView.SetShiftMode(false, false);
+
+			if (loadTasks.Count > 0)
+				await Task.WhenAll(loadTasks);
+
+			foreach (var menuItem in menuItems)
+				menuItem.Dispose();
+		}
+
+		static async Task SetMenuItemIcon(IMenuItem menuItem, ImageSource source, Context context)
+		{
+			if (source == null)
+				return;
+			var drawable = await context.GetFormsDrawable(source);
+			menuItem.SetIcon(drawable);			
+			drawable?.Dispose();
+		}
+
+		public static BottomSheetDialog CreateMoreBottomSheet(
+			Action<int, BottomSheetDialog> selectCallback, 
+			Context context,
+			List<(string title, ImageSource icon, bool tabEnabled)> items)
+		{
+			var bottomSheetDialog = new BottomSheetDialog(context);
+			var bottomSheetLayout = new LinearLayout(context);
+			using (var bottomShellLP = new LP(LP.MatchParent, LP.WrapContent))
+				bottomSheetLayout.LayoutParameters = bottomShellLP;
+			bottomSheetLayout.Orientation = Orientation.Vertical;
+			// handle the more tab
+			for (int i = 4; i < items.Count; i++)
```

Comment:
why exactly 4?

Reply:
Same as iOS

Reply:
Because that's all the bottom navigation view on android can handle if you add more than 5 items it'll crash